gravitee-apim-gateway/gravitee-apim-gateway-core/src/main/java/io/gravitee/gateway/reactive/core/context/AbstractRequest.java

```diff
@@ -197,15 +196,20 @@ public WebSocket webSocket() {
 
     @Override
     public MutableRequest contextPath(String contextPath) {
+        this.contextPath = contextPath;
+        this.pathInfo = path().substring((contextPath.length() == 1) ? 0 : contextPath.length() - 1);
+        return this;
+    }
+
+    @Override
+    public MutableRequest debugContextPath(String contextPath) {
         if (this.contextPath != null) {
+            if (!this.path.endsWith("/") && this.contextPath.endsWith("/")) {
+                this.path += "/";
+            }
             this.path = this.path.replaceFirst(this.contextPath, contextPath);
```

Comment:
This was removed from orignal method because it's seam added only for debug
https://github.com/gravitee-io/gravitee-api-management/commit/b7164124459b56fbb200d23efe06324249b293df